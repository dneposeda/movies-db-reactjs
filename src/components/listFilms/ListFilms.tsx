import React, { FC } from 'react';
import { IFilm } from "../../common/interfaces/Flims.interface";
import ShortCardFilm from '../shortCardFilm/ShortCardFilm';

type Props = {
    films: IFilm[],
}
type Component = FC<Props>;

const ListFilms: Component = ({ films }) => {
    const isEmptyList = !films?.length;
    const filmsList = isEmptyList
        ? 'Список пуст'
        : films.map((film: IFilm) => <ShortCardFilm film={ film } key={ film.id } />);

    return (
        <div className="wrapper-main">
            <div className="container">
                <div className="row">
                    { filmsList }
                </div>
            </div>
        </div>
    )
}

export default ListFilms;
