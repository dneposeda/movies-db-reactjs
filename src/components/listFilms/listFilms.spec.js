import React from 'react';
import ListFilms from './ListFilms';

const mockFilms = [
    {
        id: 2,
        title: 'Reservoir Reservoir',
        year: 2019,
        genre: 'Oscar winning Movie',
        img: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg'
    },
    {
        id: 3,
        title: 'Resedogs rvoir dogs',
        year: 2011,
        genre: 'Oscar winning Movie',
        img: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg'
    },
];

describe('ListFilms', () => {
    it('renders correctly', () => {
        const component = shallow (
            <ListFilms films={ mockFilms }/>
        );
        expect(component).toMatchSnapshot();
    });
});
