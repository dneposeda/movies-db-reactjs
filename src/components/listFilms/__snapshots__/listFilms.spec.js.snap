// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListFilms renders correctly 1`] = `
<div
  className="wrapper-main"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <ShortCartFilm
        film={
          Object {
            "genre": "Oscar winning Movie",
            "id": 2,
            "img": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
            "title": "Reservoir Reservoir",
            "year": 2019,
          }
        }
        key="2"
      />
      <ShortCartFilm
        film={
          Object {
            "genre": "Oscar winning Movie",
            "id": 3,
            "img": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
            "title": "Resedogs rvoir dogs",
            "year": 2011,
          }
        }
        key="3"
      />
    </div>
  </div>
</div>
`;
